odl/tomo/analytic/filtered_back_projection.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+from odl.discr import ResizingOperator
+from odl.trafos import FourierTransform, PYFFTW_AVAILABLE
+
+
+__all__ = ('fbp_op',)
+
+
+def fbp_op(ray_trafo, padding=True):
+    """Create filtered back-projection from a `RayTransform`.
+
+    The filtered back-projection is an approximate inverse to the ray
+    transform.
+
+    Parameters
+    ----------
+    ray_trafo : `RayTransform`
+        The ray transform (forward operator) whose approximate inverse should
+        be computed. Its geometry has to be any of the following
+        Parallel2DGeometry : Exact reconstruction
+        Parallel3dAxisGeometry : Exact reconstruction
+        FanFlatGeometry : Approximate reconstruction, correct in limit of fan
+                          angle = 0.
+        CircularConeFlatGeometry : Approximate reconstruction, correct in limit
+                                   of fan angle = 0.
+        HelicalConeFlatGeometry : Very approximate.
+
+    padding : bool, optional
+        If the data space should be zero padded. Without padding, the data may
+        be corrupted due to the circular convolution used. Using padding makes
+        the algorithm slower.
+
+    Returns
+    -------
+    fbp : `Operator`
+        Approximate inverse operator of ``ray_trafo``.
+    """
+    impl = 'pyfftw' if PYFFTW_AVAILABLE else 'numpy'
+    alen = ray_trafo.geometry.motion_params.length
+
+    if ray_trafo.domain.ndim == 2:
+        # Define ramp filter
+        def fft_filter(x):
+            return np.sqrt(x[1]**2) / (2 * alen)
+
+        # Define (padded) fourier transform
+        if padding:
+            # Define padding operator
+            ran_shp = (ray_trafo.range.shape[0],
+                       ray_trafo.range.shape[1] * 2 - 1)
```

Comment:
This is what is theoretically needed, but as you say it is hardly the case in reality. I think I'll let this stay for now though since it should be doable for most cases, worst case users can simply pick `padding=False`. If this becomes much used I'll optimize it.